### PR TITLE
Remove ButtonGroup children

### PR DESCRIPTION
**Pull request by amelako**
## Purpose

In #1167 we improved the ButtonGroup api to use the action config object with updated styles for mobile and deprecated the children prop.
In v5 we are removing the deprecated children prop from the ButtonGroup component.

## Approach and changes

- Removed the deprecated children prop from ButtonGroup component and the styles.
- Updated the ButtonGroup component usage in circuit ui to use the action prop instead of children.

## Definition of done

* [ ] Development completed
* [ ] Reviewers assigned
* [ ] Unit and integration tests
* [ ] Meets minimum browser support
* [ ] Meets accessibility requirements

### CODE REVIEW

File: packages/circuit-ui/components/ButtonGroup/ButtonGroup.tsx [hunk 1]
```diff
@@ -22,85 +22,19 @@ import Button, { ButtonProps } from '../Button';
...
-  /**
-   * @deprecated Use the `actions` prop instead.
-   */
-  children?:
-    | (ReactElement<ButtonProps> | null | undefined)[]
-    | ReactElement<ButtonProps>;
   /**
    * Direction to align the content. Either left/right
    */
   align?: 'left' | 'center' | 'right';
```
Comment (connor-baer) [inline packages/circuit-ui/components/ButtonGroup/ButtonGroup.tsx:28]: Changing the alignment is no longer supported, so this prop should be removed.

However, when trying to migrate to the new `actions` API in the web dashboard, I found many use cases where the default center alignment doesn't work. That's why I think we need to reconsider whether we can drop the alignment option.

**Comment by robinmetral** [comment]
☝️ let's park this for now while we revisit the pattern with design. Good catch @connor-baer 🙌

File: .changeset/tough-cycles-fail.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sumup/circuit-ui': major
+---
+
+Removed the deprecated `children` prop from the `ButtonGroup` component.
```
Comment (connor-baer) [inline .changeset/tough-cycles-fail.md:5]:
```suggestion
Removed the deprecated `children` prop from the `ButtonGroup` component, use the `actions` prop instead.
```

File: packages/circuit-ui/components/ButtonGroup/ButtonGroup.stories.tsx [hunk 3]
```diff
@@ -27,6 +27,7 @@ export const Base = (args: ButtonGroupProps): JSX.Element => (
 );
 
 Base.args = {
+  align: 'center',
```
Comment (connor-baer) [inline packages/circuit-ui/components/ButtonGroup/ButtonGroup.stories.tsx:30]: The current default alignment is `right`. I'd match the docs to the code.

**Review by connor-baer — APPROVED** [review]

File: packages/circuit-ui/components/Card/Card.stories.tsx [hunk 4]
```diff
@@ -108,20 +108,33 @@ export const WithFooter = () => (
...
-        </ButtonGroup>
+        <ButtonGroup
+          actions={{
+            primary: {
+              children: 'Cancel',
+            },
+            secondary: {
+              children: 'Confirm',
+            },
+          }}
```
Comment (connor-baer) [inline packages/circuit-ui/components/Card/Card.stories.tsx:119]: You flipped a couple of buttons when migrating the examples (thanks Chromatic! 🙌):

```suggestion
          actions={{
            primary: {
              children: 'Confirm',
            },
            secondary: {
              children: 'Cancel',
            },
          }}
```

File: packages/circuit-ui/components/Card/Card.stories.tsx [hunk 5]
```diff
@@ -108,20 +108,33 @@ export const WithFooter = () => (
...
+        <ButtonGroup
+          align="left"
+          actions={{
+            primary: {
+              children: 'Cancel',
+            },
+            secondary: {
+              children: 'Confirm',
+            },
+          }}
```
Comment (connor-baer) [inline packages/circuit-ui/components/Card/Card.stories.tsx:136]:
```suggestion
          actions={{
            primary: {
              children: 'Confirm',
            },
            secondary: {
              children: 'Cancel',
            },
          }}
```

File: packages/circuit-ui/components/Step/examples/YesOrNoSlider.tsx [hunk 6]
```diff
@@ -113,20 +112,20 @@ export default function YesOrNoSlider({
...
+            actions={{
+              primary: {
+                children: 'No',
+                ...getPreviousControlProps(),
+              },
+              secondary: {
+                children: 'Yes',
+                ...getNextControlProps(),
+              },
+            }}
```
Comment (connor-baer) [inline packages/circuit-ui/components/Step/examples/YesOrNoSlider.tsx:127]:
```suggestion
            actions={{
            	primary: {
                children: 'Yes',
                ...getNextControlProps(),
              },
              secondary: {
                children: 'No',
                ...getPreviousControlProps(),
              },
            }}
```

**Comment by robinmetral** [comment]
As discussed, we'll actually split this PR into two:

- a PR targeting `main` (v4 branch) will add `align` support to the `ButtonGroup` when using the `actions` API
- a PR targeting `canary` (v5 branch) will remove the deprecated `children`

This will allow us (and others) to handle migrations to the `actions` API _before_ bumping to v5 (which allows for a gradual migration, no code freezes, etc.)

**Comment by robinmetral** [comment]
> we'll actually split this PR into two:
>
> - a PR targeting main (v4 branch) will add align support to the ButtonGroup when using the actions API

👉 #1540

> - a PR targeting canary (v5 branch) will remove the deprecated children

👉 #1552

**Pull request closed without merging**